platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/dependencyConfigurationSchema.kt

```diff
@@ -89,6 +98,27 @@ class DependencyFunctionsExtractor(val configurations: DependencyConfigurations)
 }
 
 
+private
+class ImplicitDependencyCollectorFunctionExtractor(val configurations: DependencyConfigurations) : FunctionExtractor {
+    override fun memberFunctions(kClass: KClass<*>, preIndex: DataSchemaBuilder.PreIndex): Iterable<SchemaMemberFunction> = kClass.memberFunctions
```

Comment:
💭 IIRC, inspecting just `memberFunctions` would ignore properties declared in Kotlin (as their getters won't appear in `memberFunctions`.
I would suggest processing `memberProperties` as well, or, IMO less preferably, using Java reflection to scan for getters instead.